components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/RegularExpressionProtector.java

```diff
@@ -83,19 +104,93 @@ public boolean mediate(MessageContext messageContext) {
+        String tenantDomain = MultitenantUtils.getTenantDomainFromRequestURL(RESTUtils.getFullRequestPath
+                (messageContext));
+        if (StringUtils.isEmpty(tenantDomain)) {
+            tenantDomain = org.wso2.carbon.utils.multitenancy.MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+        if (!allowedTenantsList.contains(tenantDomain) &&
+                !(org.wso2.carbon.utils.multitenancy.MultitenantConstants.SUPER_TENANT_DOMAIN_NAME).equals(tenantDomain)) {
+            GatewayUtils.handleThreat(messageContext, APIMgtGatewayConstants.HTTP_SC_CODE,
+                    "This tenant is not allowed to use Regular Expression Threat Protector mediator");
+            return false;
+        } else {
+            return true;
+        }
+    }
+    /**
+     * This method returns true if the request payload size exceeds the system property
+     * 'payloadSizeLimitForRegexThreatProtector' value (in KB) defined. If this system property is not defined, this
+     * check won't be done.
+     *
+     * @param messageContext contains the message properties of the relevant API request which was
+     *                       enabled the regexValidator message mediation in flow.
+     * @return true if the payload size has exceeded the defined value in system property
+     */
+    private boolean isPayloadSizeExceeded(MessageContext messageContext) {
+        // payloadSizeLimit is in KB
+        Integer payloadSizeLimit = Integer.getInteger(APIMgtGatewayConstants.PAYLOAD_SIZE_LIMIT_FOR_REGEX_TREAT_PROTECTOR);
+        if (payloadSizeLimit == null) {
+            return false;
+        }
+        long requestPayloadSize = 0;
+        org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext) messageContext)
+                .getAxis2MessageContext();
+        Map headers = (Map) axis2MC.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+        String contentLength = (String) headers.get(HttpHeaders.CONTENT_LENGTH);
+        if (contentLength != null) {
+            requestPayloadSize = Integer.parseInt(contentLength);
+        } else {  //When chunking is enabled
+            SOAPEnvelope env = messageContext.getEnvelope();
+            if (env != null) {
+                SOAPBody soapbody = env.getBody();
+                if (soapbody != null) {
+                    byte[] size = soapbody.toString().getBytes(Charset.defaultCharset());
+                    requestPayloadSize = size.length;
+                }
+            }
+        }
+        if (requestPayloadSize > payloadSizeLimit * 1024) {
+            GatewayUtils.handleThreat(messageContext, APIMgtGatewayConstants.HTTP_SC_CODE, "Exceeded Request Payload " +
+                    "size limit allowed to be used with the enabledCheckBody option of Regular Expression Threat " +
+                    "Protector mediator");
+            return true;
+        } else {
```

Comment:
else condition not needed

Reply:
Fixed with 07871ad1b3c4d8f82ef805cbd79ad85375a05c8c